### ISSUE TITLE
Validate goal doesn't pick up formatters?

### ISSUE THREAD

**Issue by jontejj** [revelc/formatter-maven-plugin#194]
If I run the format goal it works but when i switch to validate I get "You must provide a Java or Javascript configuration file".

I.e:
```
				<executions>
					<execution>
						<goals>
							<goal>validate</goal>
						</goals>
					</execution>
				</executions>
```
doesn't work, but with "format" it does.

**Comment by hazendaz** [comment]
Thanks for reporting.  I don't have much time to look into this.  If possible, would be great if you could look into this and see if you cannot solve the issue.  What I would suggest is looking at how the normal formatting logic works and see if it's just missing something on that end.  I'll happily take pull request and can even work a release quickly to get patched.

**Comment by jontejj** [comment]
Alrighty! I might also squeeze in a feature to only validate files that
have been changed. (Another PR of course) Both as a way to make it faster
but also to introduce it slowly into a project instead of all at once.

**Comment by jontejj** [comment]
It seems like it was due to the fact that the "initialize" phase is triggered after the "validate goal. (https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html) We had this in our config setting up the path:
```
			<plugin>
                <groupId>org.commonjava.maven.plugins</groupId>
                <artifactId>directory-maven-plugin</artifactId>
                <version>0.1</version>
                <executions>
                    <execution>
                        <id>directories</id>
                        <goals>
                            <goal>directory-of</goal>
                        </goals>
                        <phase>initialize</phase>
                        <configuration>
                            <property>noxbox.rootdir</property>
                            <project>
                                <groupId>se.fortnox.noxbox</groupId>
                                <artifactId>noxbox-parent</artifactId>
                            </project>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
```

Using a hardcoded path to the config file works.

So there are two options: be happy with this the way it is and not being able to use "initialize" stuff (I think we can move our "initialize" to "validate". The other one is to switch goal to process-sources. What do you think is the best one?

**Comment by ctubbsii** [comment]
The default for this plugin is to run at "process-sources" phase.

**Comment by jontejj** [comment]
The validate goal is bound to the validate phase though.

**Comment by hazendaz** [comment]
@jontejj Yup that is bad.  We probably should move it to the process sources.  I took a look at maven again to be sure [here](https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html).  I don't see why we would want this happening so soon.  However, I suspect the generation of source files might have issues complaining that we failed to format those files.  So before we do this we are going to need to do some testing.  If we put it in initialize you will still likely run into issues but that might not occur depending on order in your pom.  Anyway, I've got something I can test that generates source but probably cannot get to this for a week or so.  For now, are you good with your work around?
